src/modules/dna-service/dna.background.ts

```diff
@@ -248,7 +253,49 @@ export class DnaService {
     return browserStorage.set(ACTION_QUEUE_KEY, []);
   }
 
+  async registerAllWallets() {
+    const isInvoked = await browserStorage.get<boolean>(
+      REGISTER_ALL_WALLETS_INVOKED_KEY
+    );
+    if (isInvoked) {
+      return;
+    }
+
+    const wallet = this.getWallet();
+    const walletGroups = await wallet.uiGetWalletGroups({
+      context: INTERNAL_SYMBOL_CONTEXT,
+    });
+
+    const ownedAddresses =
+      walletGroups
+        ?.filter((group) => !isReadonlyContainer(group.walletContainer))
+        ?.flatMap((group) =>
+          group.walletContainer.wallets.map((wallet) => ({
+            address: wallet.address,
+            origin: group.origin || undefined,
+          }))
+        ) || [];
+
+    let didFail = false;
+    for (const { address, origin } of ownedAddresses) {
+      try {
+        await this.registerWallet({
+          params: { address, origin },
+        });
+      } catch (error) {
+        didFail = true;
+        console.warn('registerWallet error', error); // eslint-disable-line no-console
+      }
+    }
+
+    if (!didFail) {
+      await browserStorage.set(REGISTER_ALL_WALLETS_INVOKED_KEY, true);
+    }
```

Comment:
I think you're setting this value even if there are no walletGroups found. What should happen in that case?

Reply:
If there are no walletGroups then we have no "unregistered wallets" to register

Reply:
Shouldn't we wait for the login event maybe?

Reply:
I subscribed to the `authenticated` event, which required me to make `account` a parameter in the `initialize` function. This is because `this.record.walletManager.groups` comes from `this.record`, and `this.record` itself is set up during the login flow: `login -> setUser -> updateCredentials -> syncWithWalletStore`. Right after this setup, the `authenticated` event is triggered